Documentation/IBAnimatable 3.0 Migration Guide.md

```diff
@@ -0,0 +1,373 @@
+
+```swift
+// Before
+view.animationType = "Pop"
+view.maskType = "Circle"
+
+// After
+view.animationType = .pop(repeatCount: 2)
+view.maskType = .circle
+```
+
+
+### Animatable UI elements
+Because `IBAnimatable` uses protocol oriented programming paradigm, all animatable UI elements, e.g. `AnimatableView`, `AnimatableButton` use the reused protocols. And we have changed the protocols to use the new enum system as mentioned above, we need to change those properties to use enum in code like:
+
+```swift
+// Before
+view.animationType = "Pop"
+view.maskType = "Circle"
+button.view.animationType = "Shake"
+
+// After
+view.animationType = .pop(repeatCount: 2)
+view.maskType = .circle
+button.animationType = .shake(repeatCount: 1)
+``` 
+
+Please find more information for [Enums](###Enums) and [Protocols](###Protocols).
+
+### Custom UI elements
+If you have created your own custom UI elements and used the protocols in `IBAnimatable` like [IBAnimatableMaterial](https://github.com/IBAnimatable/IBAnimatableMaterial) project, you need to change the implementation for some properties e.g. `animationType` and `maskType`. Here is an example for `maskType`:
+
+```swift
+class CustomView: UIView, MaskDesignable {
+	// MARK: - MaskDesignable
+  open var maskType: MaskType = .none {
+    didSet {
+      configureMask()
+      configureBorder()
+      configureMaskShadow()
+    }
+  }
+  
+  /// The mask type used in Interface Builder. **Should not** use this property in code.
+  @IBInspectable var _maskType: String? {
+    didSet {
+      maskType = MaskType(string: _maskType)
+    }
+  }
+}
+```
+
+Because Interface Builder can't support Swift enum, we have to define a property called `_maskType` as `String`. The property is `internal`, which we only use it in Interface Builder. And in the `didSet` block, we set the actual property `maskType` which is a Swift enum `MaskType`.
+
+### Interface Builder / Storyboard
+Since we have introduced new enum system, we have to change some properties like `animationType` and `maskType`. And we need to change them in the Storyboard when migrating version 3.
+
+You may see some warnings like below:
+> DemoApp.storyboard: warning: IB Designables: Ignoring user defined runtime attribute for key path "animationType" on instance of "IBAnimatable.AnimatableButton". Hit an exception when attempting to set its value: [<IBAnimatable.AnimatableButton 0x7ffc3a974800> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key animationType.
+
+Because we have change the properties name to support Interface Builder. Most of the cases, we can just add an underscore (`_`) to fix them.
+
+There are a couple of ways to fix them.
+
+0. Open the *.storyboard file with your favor editor e.g. Atom, VS Code or sublime. Replace the value as below:
```

Comment:
You can also find them by doing a search in Xcode (even doing a replace all, but that's may be dangerous)